docs/spec/special-types.rst

```diff
@@ -218,3 +214,98 @@ subtype of ``type[Base]``::
   def new_pro_user(pro_user_class: type[ProUser]):
       user = new_user(pro_user_class)  # OK
       ...
+
+.. _`union-types`:
+
+Union types
+-----------
+
+Since accepting a small, limited set of expected types for a single
```

Comment:
Now that the (draft) spec explains that types are "sets" of values, we could more properly describe unions in set theoretic terms.